### PR TITLE
Range selection

### PR DESCRIPTION
**Pull request by m-luketin**
Implemented date range selection functionality.

- Added new Advanced Calendar page to SampleApp and moved the range picker from the previous Advanced page
- Implemented selection behaviour modelled after third-party calendars
- Added Clear functionality to pop-ups in SampleApp
- Refactored some contributor code
- Handled a number of single selection bugs that arose in the process

### CODE REVIEW

File: src/Calendar.Plugin/Shared/Controls/MonthDaysView.xaml.cs [hunk 1]
```diff
@@ -342,15 +377,13 @@ protected override void OnPropertyChanged([CallerMemberName] string propertyName
...
-                    break;
-
                 case nameof(DisplayedMonthYear):
                 case nameof(Events):
                 case nameof(MinimumDate):
                 case nameof(MaximumDate):
                 case nameof(OtherMonthDayIsVisible):
-                    UpdateDays(AnimateCalendar);
+                case nameof(RangeSelectionStartDate):
+                case nameof(RangeSelectionEndDate):                    UpdateDays(AnimateCalendar);
```
Comment (antonioseric) [inline src/Calendar.Plugin/Shared/Controls/MonthDaysView.xaml.cs:386]: UpdateDays should be in new line

**Review by antonioseric — CHANGES_REQUESTED** [review]
Also don't forget to add images for Ranged and documentation in Readme.md

File: src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedPageViewModel.cs [hunk 2]
```diff
@@ -25,7 +25,20 @@ public class AdvancedPageViewModel : BasePageViewModel, INotifyPropertyChanged
...
             {
-                string message = calendarPickerResult.IsSuccess ? $"Received date from popup: {calendarPickerResult.SelectedDate:dd/MM/yy}" : "CalendarPicker Canceled!";
+                string message = calendarPickerResult.IsSuccess ? $"Received date from popup: {calendarPickerResult.SelectedDate:dd/MM/yy}" : "Calendar Picker Canceled!";
+
+                await App.Current.MainPage.DisplayAlert("Popup result", message, "Ok");
+                Console.WriteLine(message);
+            }));
+        });
+
+        public ICommand OpenRangePickerCommand => new Command(async () =>
```
[the remark this answers is not in the document]
Comment (m-luketin) [reply src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedPageViewModel.cs:35]: Definitely, removed now.

File: README.md [hunk 3]
```diff
@@ -24,6 +24,11 @@ If you are coming back take a look on the [Changelog here](https://github.com/li
 | ------- | ------ |
 | ![Android Custom Calendar Screenshot](https://github.com/lilcodelab/Xamarin.Plugin.Calendar/blob/master/art/android-advanced.png) | ![iPhone Custom Calendar Screenshot](https://github.com/lilcodelab/Xamarin.Plugin.Calendar/blob/master/art/iphone-advanced.png) |
 
+## Range implementation
+| Android | iPhone |
+| ------- | ------ |
+| ![Android Custom Calendar Screenshot](https://github.com/lilcodelab/Xamarin.Plugin.Calendar/blob/master/art/android-range.png) | ![iPhone Custom Calendar Screenshot](https://github.com/lilcodelab/Xamarin.Plugin.Calendar/blob/master/art/iphone-range.png) |
+
```
Comment (antonioseric) [inline README.md:32]: Documentation for new property is still missing, please add i markdown new section how to use/turn on this feature

File: src/Calendar.Plugin/CalendarPlugin - Backup.csproj [hunk 4]
```diff
@@ -0,0 +1,101 @@
+﻿<Project Sdk="MSBuild.Sdk.Extras/2.1.2">
```
Comment (antonioseric) [inline src/Calendar.Plugin/CalendarPlugin - Backup.csproj:1]: this needs to be removed

Comment (antonioseric) [inline src/Calendar.Plugin/CalendarPlugin - Backup.csproj:1]: This should be removed

File: src/Calendar.Plugin/CalendarPlugin.csproj [hunk 5]
```diff
@@ -1,8 +1,7 @@
-﻿<Project Sdk="MSBuild.Sdk.Extras/2.0.54">
+﻿<Project Sdk="MSBuild.Sdk.Extras/2.1.2">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;MonoAndroid81;Xamarin.iOS10</TargetFrameworks>
-    <Authors>Josip Caleta</Authors>
+    <TargetFrameworks>netstandard2.0;MonoAndroid10.0;MonoAndroid90;Xamarin.iOS10</TargetFrameworks>
```
[the remark this answers is not in the document]
Comment (antonioseric) [reply src/Calendar.Plugin/CalendarPlugin.csproj:4]: this is still there

File: src/Calendar.Plugin/CalendarPlugin.csproj [hunk 6]
```diff
@@ -1,8 +1,7 @@
-﻿<Project Sdk="MSBuild.Sdk.Extras/2.0.54">
+﻿<Project Sdk="MSBuild.Sdk.Extras/2.1.2">
```
Comment (antonioseric) [inline src/Calendar.Plugin/CalendarPlugin.csproj:1]: please change version to `3.0.23`

File: src/Calendar.Plugin/Shared/Enums/SelectionType.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
+﻿namespace Xamarin.Plugin.Calendar.Enums
+{
+    /// <summary>
+    /// Set type of event indicator on calendar
+    /// </summary>
+    public enum SelectionType
+    {
+        /// <summary>
+        /// Day title maximum length of one character.
```
Comment (antonioseric) [inline src/Calendar.Plugin/Shared/Enums/SelectionType.cs:9]: comments are false

File: README.md (hunk 3 above)
Comment (antonioseric) [reply README.md:32]: also change Supported versions, forms to 4.5+ and android to 10.0+ and ios 9.0+

**Review by antonioseric — CHANGES_REQUESTED** [review]

File: src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedRangeSelectionPageViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+        public ICommand OpenRangePickerCommand => new Command(async () =>
+        {
+            await PopupNavigation.Instance.PushAsync(new CalendarRangePickerPopup(async (calendarPickerResult) =>
+            {
+                string message = calendarPickerResult.IsSuccess ?
+                    $"Received date range from popup: {calendarPickerResult.SelectedStartDate:dd.MM.yyyy} - {calendarPickerResult.SelectedEndDate:dd.MM.yyyy}" :
+                    "Calendar Range Picker Canceled!";
+
+                await App.Current.MainPage.DisplayAlert("Popup result", message, "Ok");
+                Console.WriteLine(message);
```
Comment (bbazina) [inline src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedRangeSelectionPageViewModel.cs:30]: this should be removed

File: src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedRangeSelectionPageViewModel.cs [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+        {
+            get => _selectedEndDate;
+            set => SetProperty(ref _selectedEndDate, value);
+        }
+
+        private static async Task DayTapped(DateTime date)
+        {
+            var message = $"Received tap event from date: {date}";
+            await App.Current.MainPage.DisplayAlert("DayTapped", message, "Ok");
+            Console.WriteLine(message);
```
Comment (bbazina) [inline src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedRangeSelectionPageViewModel.cs:147]: this should be removed

File: src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedRangeSelectionPageViewModel.cs [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+        }
+
+        private async Task ExecuteEventSelectedCommand(object item)
+        {
+            if (item is AdvancedEventModel eventModel)
+            {
+                var title = $"Selected: {eventModel.Name}";
+                var message = $"Starts: {eventModel.Starting:HH:mm}{Environment.NewLine}Details: {eventModel.Description}";
+                await App.Current.MainPage.DisplayAlert(title, message, "Ok");
+                Console.WriteLine(message);
```
Comment (bbazina) [inline src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedRangeSelectionPageViewModel.cs:157]: this should be removed

**Review by antonioseric — APPROVED** [review]

**Pull request merged**